tools/distpartitioning/dist_lookup.py

```diff
@@ -75,8 +85,12 @@ def __init__(self, input_dir, ntype_names, id_map, rank, world_size):
                 end = count
             type_nid_begin.append(start)
             type_nid_end.append(end)
+
             # Slice the partition-ids which belong to the current instance.
-            partid_list.append(ntype_partids[start:end])
+            partid_list.append(copy.deepcopy(ntype_partids[start:end]))
```

Comment:
are you sure that only use `deepcopy` and `del` together, occupied memory can be released? `del` is not sufficient?

Reply:
This is the reason for these 2 lines of code. 

Earlier, partition-id file was read and only a slice of this numpy array was stored in the dist. lookup service object. The problem is that since we are storing a slice of the numpy array, entire numpy array is stored in the memory/RAM and this was not released until the end of execution of the pipeline. Since these files are very large and can be 50GB in memory.... so this is huge chunk of memory which is used unnecessarily. 

To solve the above problem now, we do the following: 
1. after reading the partition-id file, we copy the slice of the numpy array which is owned by the current process. In this way the original numpy array which was created during this file is read will be released once the numpy array becomes out of scope. 
2. For copying purposes, we use copy.deepcopy() method. This will allocate new block of memory and copies the slice of numpy array.... so that no references of the numpy array are stored or referred anywhere. 
3. We also explicitly delete the memory by using `del` keyword for the numpy array variable. 
4. Per python documentation, `del variablename` or variable_name = None either of these ways will deallocate the memory (with all the references) behind the numpy array. 

In my testing with Fe_large graph, the partition id files can occupy 32GB of physical memory, which is RAM. By doing the deepcopy and explicitly releasing the memory by using `del` operator releases this memory. Please also note that <variable_name> = None will release the memory of the numpy array. 

Do you think there is a better way of handling this problem. ?? please let me know.